src/directives/mmsDiffAttr.js

```diff
@@ -17,212 +17,166 @@ angular.module('mms.directives')
  * @description
  *  Compares a element at two different refs/commits and generates a pretty diff.
  * ## Example
- * <mms-diff-attr mms-base-element-id="" (mms-compare-element-id="") mms-attr="name|doc|val"
- * (mms-base-project-id="" mms-compare-project-id="" mms-base-ref-id="" mms-compare-ref-id=""
+ * <mms-diff-attr mms-element-id="" mms-attr="name|doc|val"
+ * (mms-project-id="" mms-base-ref-id="" mms-compare-ref-id=""
  * mms-base-commit-id="" mms-compare-commit-id="")></mms-diff-attr>
- *
- * @param {string} mmsBaseElementId The id of the element to do comparison of
- * @param {string} mmsAttr Attribute to use -  ie `name`, `doc` or `value`
- * @param {string} mmsBaseProjectId Base project ID for original/base element
- * @param {string} mmsCompareProjectId Compare project ID for compare element
- * @param {string=master} mmsBaseRefId Base ref ID or master, defaults to current ref or master
- * @param {string=master} mmsCompareRefId Compare ref ID or master, defaults to base ref ID
- * @param {string=latest} mmsBaseCommitId Base commit id, default is latest
- * @param {string=latest} mmsCompareCommitId Compare commit id, default is latest
  */
-function mmsDiffAttr(ElementService, $compile, $rootScope, $interval, $templateCache) {
+function mmsDiffAttr($compile, $rootScope, $interval, $templateCache, $q, ElementService) {
     var template = $templateCache.get('mms/templates/mmsDiffAttr.html');
 
-    var mmsDiffAttrLink = function(scope, element, attrs, mmsViewCtrl) {
-        var ran = false;
-        var viewOrigin;
+    return {
```

Comment:
just to clarify, the history pane isn't the only place the diff can be used, it's just what ve was using. users can fill in the other parameters inside a view or from md, etc. The user guide had listed the other params as part of the interface, and anything not supplied is derived or has a default (if viewCtrl is present, base projectId and base ref Id are derived from it, if compare params are not present, they'll be the same as base)

Reply:
i took them out because i thought it is only used in history pane and couldn't see why it has to be like that:
1. for projectId: since we can't compare across projects, i think we just need one project id
2. for elementId: we only want to compare the same element. Thus, only one id is needed.
3. baseRefId: I will add additional default in viewOrigin.refId is present


Or must mms-diff-attr directive allows for all those extras params for the existing view/md to work? I am not entirely sure how it is being used in view/md.

Reply:
yea..the diff directive can be used by users inside a view - however they write it (pure html, through a script, etc), we documented the api in the user guide so it's more flexible as a general diff between two texts. it can happen because elements can be moved across projects as time goes on, or an element can be refactored with a different id, etc

Reply:
ah okay. I'll revert it then